ironfish/src/wallet/wallet.ts

```diff
@@ -425,7 +453,11 @@ export class Wallet {
 
     for (const [accountId, decryptedNotes] of decryptedNotesByAccountId) {
       const account = this.accounts.get(accountId)
-      Assert.isNotUndefined(account, `syncTransaction: No account found for ${accountId}`)
+
+      if (!account) {
+        this.logger.warn(`Account ${accountId} removed from wallet during note decryption`)
```

Comment:
warnings are for things going wrong that aren't fatal, but why is it wrong for someone to remove a wallet?

Reply:
Fair point; it's may be 'wrong' from the perspective of the programmer who expects the account to exist, but not from the perspective of the user who removed the account

I'll remove the warning